packages/react-form/src/hooks/list/test/baselist.test.tsx

```diff
@@ -768,4 +768,126 @@ describe('useBaseList', () => {
+            </>
+          ))}
+
+          {defaultValue.map(variants => (
+            <>
+              <p>Default: {variants.price}</p>
+              <p>Default: {variants.optionName}</p>
+              <p>Default: {variants.optionValue}</p>
+            </>
+          ))}
+
+          <button type="button" onClick={onNewDefault} />
+        </>
+      );
+    }
+
+    it('returns the value of the baselist', () => {
+      const price = '1.00';
+      const optionName = 'material';
+      const optionValue = 'cotton';
+      const variants: Variant[] = [
+        {
+          price,
+          optionName,
+          optionValue,
+        },
+      ];
+
+      const wrapper = mount(<TestListWithValue list={variants} />);
+      expect(wrapper).toContainReactText(`Value: ${price}`);
+      expect(wrapper).toContainReactText(`Value: ${optionName}`);
+      expect(wrapper).toContainReactText(`Value: ${optionValue}`);
+    });
+
+    it('sets and returns the default value of the base list', () => {
+      const price = '1.00';
+      const optionName = 'material';
+      const optionValue = 'cotton';
+      const variants: Variant[] = [
+        {
+          price,
+          optionName,
+          optionValue,
+        },
+      ];
+      const newDefaultPrice = '2.00';
+      const newDefaultOption = 'color';
+      const newDefaultOptionValue = 'blue';
+
+      const wrapper = mount(<TestListWithValue list={variants} />);
+
+      expect(wrapper).toContainReactText(`Default: ${price}`);
+      expect(wrapper).toContainReactText(`Default: ${optionName}`);
+      expect(wrapper).toContainReactText(`Default: ${optionValue}`);
+
+      wrapper.find('button')!.trigger('onClick', [
+        {
+          price: newDefaultPrice,
+          optionName: newDefaultOption,
+          optionValue: newDefaultOptionValue,
+        },
+      ]);
+
+      expect(wrapper).toContainReactText(`Default: ${newDefaultPrice}`);
+      expect(wrapper).toContainReactText(`Default: ${newDefaultOption}`);
+      expect(wrapper).toContainReactText(`Default: ${newDefaultOptionValue}`);
+    });
+
+    it('can reinitialize after newDefaultValue is called', () => {
+      const variants: Variant[] = [
+        {
+          price: '1.00',
+          optionName: 'material',
+          optionValue: 'cotton',
+        },
+      ];
+
+      const newDefaultPrice = '2.00';
+      const newDefaultOption = 'color';
+      const newDefaultOptionValue = 'blue';
+
+      const wrapper = mount(<TestListWithValue list={variants} />);
+
+      wrapper.find('button')!.trigger('onClick', [
+        {
+          price: newDefaultPrice,
+          optionName: newDefaultOption,
+          optionValue: newDefaultOptionValue,
+        },
+      ]);
+
+      expect(wrapper).toContainReactText(`Default: ${newDefaultPrice}`);
+      expect(wrapper).toContainReactText(`Default: ${newDefaultOption}`);
+      expect(wrapper).toContainReactText(`Default: ${newDefaultOptionValue}`);
```

Comment:
I would remove the three expect and inline the "new" values. This part is already tested above. Here we want to test the `useEffect` that reinitializes the list when a new config is passed.